lib/browser/api/web-contents.js

```diff
@@ -326,6 +326,17 @@ WebContents.prototype._init = function () {
     }
   })
 
+  this.on('-ipc-invoke', function (event, channel, args) {
+    event.reply = (result) => event.sendReply({ result })
+    event.throw = (error) => event.sendReply({ error: error.toString() })
+    this.emit('ipc-invoke', event, channel, ...args)
```

Comment:
What is the purpose of this `emit`?

Reply:
just following other handlers. it could be useful to blanket catch all invokes? but maybe it should just be removed.

Reply:
The others are documented here --> https://electronjs.org/docs/api/web-contents#event-ipc-message and are used to as you say, catch all the IPC.

Not sure that in the case of invokes where we expect a 1:1 relationship between message and handler it makes sense.  Up to you though 👍 Either remove or document as API with the others

Reply:
i'm going to remove it for now in favour of a smaller API surface. if we add it back in future we should consider things like:
- should the catch-all handler be able to e.g. `event.preventDefault` to prevent the registered handler from executing?
- should this be implemented as an event on the webcontents or an event on ipcMain?
- if the latter, should it be an event or should it be something like `setDefaultInvokeHandler`?

etc.